lib/python/qmk/json_encoders.py

```diff
@@ -71,22 +80,43 @@ def indent_str(self):
 class InfoJSONEncoder(QMKJSONEncoder):
     """Custom encoder to make info.json's a little nicer to work with.
     """
-    def encode_dict(self, obj):
+    def encode_dict(self, obj, key):
         """Encode info.json dictionaries.
         """
         if obj:
-            if set(("x", "y")).issubset(obj.keys()):
-                # These are part of a layout/led_config, put them on a single line.
-                return "{ " + ", ".join(f"{self.encode(key)}: {self.encode(element)}" for key, element in sorted(obj.items())) + " }"
-
-            else:
-                self.indentation_level += 1
-                output = [self.indent_str + f"{json.dumps(key)}: {self.encode(value)}" for key, value in sorted(obj.items(), key=self.sort_dict)]
-                self.indentation_level -= 1
-                return "{\n" + ",\n".join(output) + "\n" + self.indent_str + "}"
+            self.indentation_level += 1
+            output = [self.indent_str + f"{json.dumps(k)}: {self.encode(v, k)}" for k, v in sorted(obj.items(), key=self.sort_dict)]
+            self.indentation_level -= 1
+            return "{\n" + ",\n".join(output) + "\n" + self.indent_str + "}"
         else:
             return "{}"
 
+    def sort_layout(self, key):
+        key = key[0]
+
+        if key == 'label':
+            return '00label'
+
+        elif key == 'x':
+            return '01x'
+
+        elif key == 'y':
+            return '02y'
+
+        elif key == 'w':
+            return '03w'
+
+        elif key == 'h':
+            return '04h'
+
+        elif key == 'flags':
+            return '05flags'
+
+        elif key == 'matrix':
+            return '06matrix'
```

Comment:
How strongly do you prefer this sorting order? Would it be better to put matrix first, given that the firmware-specific stuff actually cares about that, compared to any of the others?

Reply:
The order I've been using for my work is `label`, `matrix`, `x`, `y`, `w`, `h`.